lib/iris/tests/unit/fileformats/netcdf/test_save.py

```diff
@@ -59,6 +59,17 @@ def test_attributes_arrays(self):
             ds.close()
         self.assertArrayEqual(res, np.arange(2))
 
+    def test_no_special_attribute_clash(self):
+        c1 = Cube([0], var_name='test', attributes={'name': 'bar'})
+        c2 = Cube([0], var_name='test_1', attributes={'name': 'bar_1'})
```

Comment:
I need it to reproduce the problem - I've not looked at the NetCDF code in detail, but I think it needs to be trying to save a CubeList to hit the setattr lines. Presumably the attributes get set as global if it's saving one Cube, but variable attributes if it's saving many?

Happy to add a comment to that effect if you think it needs it.